main.cpp

```diff
@@ -20,17 +22,34 @@ int main() {
 	sf::RenderWindow window(sf::VideoMode(SCREEN_WIDTH, SCREEN_HEIGHT), GAME_TITLE);
 	window.setFramerateLimit(60);	//Limit klatek do 60
 	
-	sf::RectangleShape rec = sf::RectangleShape(sf::Vector2f(50.0f, 50.0f));
+	sf::RectangleShape rec = sf::RectangleShape(sf::Vector2f(50.0f*4, 37.0f*4));
```

Comment:
Czemu tutaj mnożysz tak na twardo? Jeżeli to jest jakiś mnożnik to powinien być zawarty w zmiennej imo.

Reply:
Nie wiem co z tym w sumie począć. Generalnie jeśli stworzysz obiekt gracza o takich wymiarach jak textura, w tym przypadku 50x37 pixeli, to on jest praktycznie niewidoczny. Dlatego pomnożyłem x4. Myślałem, że po prostu będzie jakaś #definicja, która ustawi jakąś skalę. Ale przecież różne obiekty mogą mieć różne skalowanie, więc to powinna być jakaś właściwość rysowanego obiektu. Ale GameObject jako jeden z parametrów otrzymuje sf::Shape, więc skala musi być już gdzieś ustawiona, w momencie tworzenia, w tym przypadku - sf::RectangleShape.

Reply:
Można zrobić tak, żeby na początku wczytywać dane o ekranie, po czym obliczać jakiś mnożnik, dzięki któremu wartości wysokości i szerokości były w procentach. Np:
screenX, screenY = getScreenDimensions()
scale = screenY;
shape = new RectangleShape( sf::Vector2f( 20 * scale, 20 * scale) );
//shape ma wymiary 20% wysokości ekranu w osi X i 20% wysokości ekranu w osi Y.